lib/assertions/exception.js

```diff
@@ -43,7 +43,13 @@ module.exports = function(referee) {
             }
 
             if (typeof matcher === "object" && !samsam.match(err, matcher)) {
-                return this.fail(typeFailMessage);
+                var prop;
+                for (prop in matcher) {
```

Comment:
This will also iterate on properties in the prototype chain, is that intentional?
If it is, then I think it would be good to leave a comment, so people that are trained to look for this as a mistake (me) won't be triggered.
Otherwise, I'd suggest using either `Object.keys(matcher).forEach` or `hasOwnProperty` to ensure that we don't also examine properties from the prototype chain.

Reply:
I did it this way, because this is how `samsam.match` iterates the object. Do you think it's safe to do it differently? Because I prefer only iterating own properties as well.

Reply:
We'd have to do another major to samsam if we change that I guess ... do you want to do that now, or leave it for some uncertain future?